### ISSUE TITLE
Some overview page links need accessibility improvements

### ISSUE THREAD

**Issue by carmacleod** [carbon-design-system/design-system-website#262]
The Getting Started, Style, Components and Resources links on the overview page all have a redundant icon link. This adds unnecessary tabbing for keyboard users, and it adds unnecessary extra reading for screen reader users.

Here's a screen snap of the 4 + 4 = 8 links:

![image](https://user-images.githubusercontent.com/3331913/41005361-1aba7568-68eb-11e8-8934-c09997a03f3b.png)

The simple (but incomplete) fix is to add `tabindex="-1"` to the `list-item__icon` anchor link elements, so that the extra 4 icon links are taken out of the tab order.

The better fix would require some css rework. The icon, text, and description should all be one link. Because that's what it really is - semantically. This would have the added advantage that mouse users would have a larger click target. Keyboard users would only need to type tab once per link, and screen reader users could bring up a list of links that doesn't have the repetition. For example, the current list of these links looks like this:
Getting Started
Getting Started
Style
Style
Components
Components
Resources
Resources

Here's what the html might look like (I don't know how to make this work using `<Link>` instead):
```
<li class="overview-page__list-item">
  <a href="/getting-started" class="list-item__link" aria-labelledby="getting-started-info__label" aria-describedby="getting-started-info__description">
    <img class="list-item__icon" src="/images/getting-started.png" alt="">
    <div class="list-item__info">
      <span id="getting-started-info__label">Getting Started</span>
      <p id="getting-started-info__description">Onboarding for designers and developers who are using Carbon for the first time.</p>
    </div>
  </a>
</li>
```

**Comment by carmacleod** [comment]
I guess they're a lot like "Clickable Tiles".

**Comment by carmacleod** [comment]
On a similar topic, the 4 clickable tiles on the overview page: Carbon Design Kit, Carbon Components, Carbon CodePen, and Theming Sandbox have a slightly different problem. Looking at the list of links (screen snap from NVDA screen reader, below) we can see that all of the content of the clickable tile (aka link) are being used as the link's label.

The problem here is that the label text and the description text are both being read when the user arrives at the link, and this is too verbose. The preferred way is for the anchor (link) to have `aria-labelledby="carbon-design-kit__info__label" aria-desribedby="carbon-design-kit__info__description"` and put those 2 id's on the 2 paragraph elements in the tile.

![image](https://user-images.githubusercontent.com/3331913/41057411-0fe5112a-6995-11e8-8756-0edbbdf9c850.png)
